mcs/class/corlib/System.Reflection.Emit/ModuleBuilder.cs

```diff
@@ -376,44 +369,33 @@ public override Type GetType (string className, bool throwOnError, bool ignoreCa
 			if (className.Length == 0)
 				throw new ArgumentException ("className");
 
-			int subt;
-			string orig = className;
-			string modifiers;
 			TypeBuilder result = null;
 
 			if (types == null && throwOnError)
 				throw new TypeLoadException (className);
 
-			subt = className.IndexOfAny (type_modifiers);
-			if (subt >= 0) {
-				modifiers = className.Substring (subt);
-				className = className.Substring (0, subt);
-			} else
-				modifiers = null;
+			TypeSpec ts = TypeSpec.Parse (className);
 
 			if (!ignoreCase) {
-				result =  name_cache [className] as TypeBuilder;
+				var displayNestedName = ts.WithoutModifiers().TypeName;
```

Comment:
I think this could be accomplished in a much more straightforward way than this setup that will:

1)create a new TypeSpec
2)create a new TypeSpecTypeName
3)use TypeSpecTypeName just for this purpose

How about a GetDisplayFullName variant that ignores modifiers? It would avoid all the intermediate steps.